packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue

```diff
@@ -111,32 +119,45 @@ const isEmpty = computed(() => props.options.length === 0);
       <OnyxEmpty>{{ t("empty") }}</OnyxEmpty>
     </slot>
 
-    <ul v-else v-scroll-end v-bind="listbox" class="onyx-listbox__options">
-      <OnyxListboxOption
-        v-for="option in props.options"
-        :key="option.id.toString()"
+    <div v-else v-scroll-end v-bind="listbox" class="onyx-listbox__wrapper">
+      <ul
+        v-for="([group, options], index) in Object.entries(groupedOptions)"
+        :key="index"
+        class="onyx-listbox__group"
         v-bind="
-          headlessOption({
-            value: option.id,
-            label: option.label,
-            disabled: option.disabled,
-            selected: option.id === props.modelValue,
+          headlessGroup({
+            label: group,
           })
         "
-        :active="option.id === activeOption"
       >
-        {{ option.label }}
-      </OnyxListboxOption>
-
-      <li v-if="props.lazyLoading?.loading" class="onyx-listbox__slot">
-        <OnyxLoadingIndicator class="onyx-listbox__loading" />
-      </li>
-
-      <li v-if="slots.optionsEnd" class="onyx-listbox__slot">
-        <slot name="optionsEnd"></slot>
-      </li>
-    </ul>
-
+        <span v-if="group != ''" class="onyx-listbox__group-name onyx-text--small">{{
+          group
+        }}</span>
+        <OnyxListboxOption
+          v-for="option in options as any"
+          :key="option.id.toString()"
+          v-bind="
+            headlessOption({
+              value: option.id,
+              label: option.label,
+              disabled: option.disabled,
+              selected: option.id === props.modelValue,
+            })
+          "
+          :active="option.id === activeOption"
+        >
+          {{ option.label }}
+        </OnyxListboxOption>
+
+        <li v-if="props.lazyLoading?.loading" class="onyx-listbox__slot">
+          <OnyxLoadingIndicator class="onyx-listbox__loading" />
+        </li>
+
+        <li v-if="slots.optionsEnd" class="onyx-listbox__slot">
+          <slot name="optionsEnd"></slot>
+        </li>
```

Comment:
This is currently displayed after every option but it should only be placed at the very end of all options